plugins/node/opentelemetry-instrumentation-graphql/src/instrumentation.ts

```diff
@@ -390,29 +389,52 @@ export class GraphQLInstrumentation extends InstrumentationBase {
     });
   }
 
+  private getOperationType(
+    operation: graphqlTypes.DefinitionNode | undefined
+  ): string | null {
+    if (operation && operation.kind == 'OperationDefinition') {
+      return operation.operation;
+    }
+
+    return null;
+  }
+
+  private getOperationName(
+    operation: graphqlTypes.DefinitionNode | undefined,
+    processedArgs: graphqlTypes.ExecutionArgs
+  ): string | null {
+    if (operation && operation.kind == 'OperationDefinition') {
+      if (operation.name) {
+        return operation.name.value;
+      }
+    }
+    if (processedArgs.operationName) {
+      return processedArgs.operationName;
+    }
+    return null;
+  }
+
   private _createExecuteSpan(
     operation: graphqlTypes.DefinitionNode | undefined,
     processedArgs: graphqlTypes.ExecutionArgs
   ): api.Span {
     const config = this._getConfig();
 
-    const span = this.tracer.startSpan(SpanNames.EXECUTE, {});
-    if (operation) {
-      const name = (operation as graphqlTypes.OperationDefinitionNode)
-        .operation;
-      if (name) {
-        span.setAttribute(AttributeNames.OPERATION, name);
-      }
-    } else {
-      let operationName = ' ';
-      if (processedArgs.operationName) {
-        operationName = ` "${processedArgs.operationName}" `;
+    const operationType = this.getOperationType(operation);
+    const operationName = this.getOperationName(operation, processedArgs);
+
+    const span = this.tracer.startSpan(createExecuteSpanName());
+
+    if (operationType) {
+      span.setAttribute(AttributeNames.OPERATION_TYPE, operationType);
+      span.updateName(createExecuteSpanName(operationType));
+    }
+
+    if (operationName) {
+      span.setAttribute(AttributeNames.OPERATION_NAME, operationName);
+      if (operationType) {
+        span.updateName(createExecuteSpanName(operationType, operationName));
       }
-      operationName = OPERATION_NOT_SUPPORTED.replace(
```

Comment:
I have removed this, as i don't believe it's needed to be formatted this way. Maybe we should rather add an attribute that would represent that the operation is "not supported"?

Reply:
why do you think so ?
currently it will help to identify the operation name that is not supported, after yr change it will not be possible, so I see here a regression instead of improvement.

Reply:
That is true, but making `operationName` to be `Operation ${operationName} not supported` breaks what the "opeationName" field is supposed to be.

I would stick with "opertationName" to represent the name so the value is more predictable. Especially if you want to use it with some automation, where with current implementation you would have to parse the string to get out the actual operation name.

Maybe adding `operationUnsupported: true` or something in this sense, to indicate that operation is not supported?